kernel/syscall.c

```diff
@@ -8,7 +8,13 @@
 #include "threads.h"
 #include "vga.h"
 
-typedef void (*syscall_func) (void);
+typedef void __attribute__((noreturn)) (*syscall_func) (void);
```

Comment:
Does this actually typecheck that the functions in syscall_defns are all noreturn? if so, cool!

Reply:
It does, apparently!  I didn't even think about the possibility of that not working, so that's a relief.

kernel/syscall.c:65:3: warning: initialization makes '__attribute__((noreturn))' qualified function pointer from unqualified [-Wdiscarded-qualifiers]

Reply:
neat!